app/screens/channel_notification_preference/channel_notification_preference.js

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {intlShape} from 'react-intl';
+import {
+    Platform,
+    ScrollView,
+    View,
+} from 'react-native';
+
+import {ViewTypes} from '@constants';
+
+import StatusBar from 'app/components/status_bar';
+import SectionItem from 'app/screens/settings/section_item';
+import {preventDoubleTap} from 'app/utils/tap';
+import {alertErrorWithFallback} from 'app/utils/general';
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+import FormattedText from 'app/components/formatted_text';
+import {t} from '@utils/i18n';
+
+export default class ChannelNotificationPreference extends PureComponent {
+    static propTypes = {
+        actions: PropTypes.shape({
+            updateChannelNotifyProps: PropTypes.func.isRequired,
+        }),
+        channelMember: PropTypes.object.isRequired,
+        theme: PropTypes.object.isRequired,
+        isLandscape: PropTypes.bool.isRequired,
+    };
+
+    static contextTypes = {
+        intl: intlShape.isRequired,
+    };
+
+    constructor(props) {
+        super(props);
+
+        const channelNotifyProps = props.channelMember && props.channelMember.notify_props;
+
+        this.state = {
+            notificationLevel: channelNotifyProps.push || ViewTypes.NotificationLevels.DEFAULT,
+        };
+    }
+
+    handlePress = preventDoubleTap(async (newNotificationLevel) => {
+        const {actions, channelMember} = this.props;
+        const {notificationLevel} = this.state;
+
+        if (newNotificationLevel === notificationLevel) {
+            // tapped on current selection.
+            return;
+        }
+
+        this.setState({
+            notificationLevel: newNotificationLevel,
+        });
+
+        const props = {push: newNotificationLevel};
+
+        const {error} = await actions.updateChannelNotifyProps(channelMember.user_id, channelMember.channel_id, props);
+        if (error) {
+            const {intl} = this.context;
+            alertErrorWithFallback(
+                intl,
+                error,
+                {
+                    id: t('channel_notifications.preference.save_error'),
+                    defaultMessage: "We couldn't save notification preference. Please check your connection and try again.",
```

Comment:
Maybe this error should read differently, something more generic without actually mentioning connectivity  issues as we don't exactly  know what went wrong here. @matthewbirtch any suggestions?